modules/theme-tools/content-options/featured-images-fallback.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * Get one image from a specified post in the following order:
+ * Featured Image then first image from the_content HTML
+ * and filter the post_thumbnail_html
+ *
+ * @param string       $html              The HTML for the image markup.
+ * @param int          $post_id           The post ID to check.
+ * @param int          $post_thumbnail_id The ID of the featured image.
+ * @param string       $size              The image size to return, defaults to 'post-thumbnail'.
+ * @param string|array $attr              Optional. Query string or array of attributes.
+ *
+ * @return string      $html              Thumbnail image with markup.
+ */
+function jetpack_featured_images_fallback_get_image( $html, $post_id, $post_thumbnail_id, $size, $attr ) {
+	if ( ! empty( $html ) || (bool) 1 !== (bool) get_option( 'jetpack_content_featured_images_fallback', 1 ) ) {
+		return trim( $html );
+	}
+
+	if ( jetpack_featured_images_should_load() ) {
+		$opts = jetpack_featured_images_get_settings();
+
+		if ( ( true === $opts['archive'] && ( is_home() || is_archive() || is_search() ) && ! $opts['archive-option'] )
+			|| ( true === $opts['post'] && is_single() && ! $opts['post-option'] ) ) {
+				return trim( $html );
+		}
+	}
+
+	if ( class_exists( 'Jetpack_PostImages' ) ) {
+		global $_wp_additional_image_sizes;
+
+		$args = array(
+			'from_thumbnail'  => false,
+			'from_slideshow'  => true,
+			'from_gallery'    => true,
+			'from_attachment' => false,
+		);
+
+		$image = Jetpack_PostImages::get_image( $post_id, $args );
+
+		if ( ! empty( $image ) ) {
+			$image['width']  = '';
+			$image['height'] = '';
+			$image['crop']   = '';
+
+			if ( array_key_exists( $size, $_wp_additional_image_sizes ) ) {
+				$image['width']  = $_wp_additional_image_sizes[ $size ]['width'];
+				$image['height'] = $_wp_additional_image_sizes[ $size ]['height'];
+				$image['crop']   = $_wp_additional_image_sizes[ $size ]['crop'];
+			}
+
+			$image_src = Jetpack_PostImages::fit_image_url( $image['src'], $image['width'], $image['height'] );
+
+			// Use the theme's crop setting rather than forcing to true
+			$image_src = add_query_arg( 'crop', $image['crop'], $image_src );
```

Comment:
Yes, most likely. Although for some reason the fallback doesn't seem to work at all in my tests with Button 2. I must be missing something. Could you walk me through the steps to test this?

Here is what I tried:

1. Check out the branch.
2. Install the Button 2 theme from WordPress.com.
3. In the customizer, enable the 3 last options for Featured Images.
4. Save and Publish.
5. Publish a post including a large image attached to the post and inserted in the post content, but not set as Featured Image.

Shouldn't that image then appear as a Featured Image in the theme, right above the post title?